vncviewer/MonitorIndicesParameter.cxx

```diff
@@ -0,0 +1,209 @@
+/* Copyright 2021 Hugo Lundin <huglu@cendio.se> for Cendio AB.
+ * 
+ * This is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ * 
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this software; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307,
+ * USA.
+ */
+
+#include <algorithm>
+#include <vector>
+#include <string>
+#include <limits>
+#include <set>
+#include <stdlib.h>
+#include <stdexcept>
+
+#include "i18n.h"
+#include <FL/Fl.H>
+#include <rfb/LogWriter.h>
+
+#include "MonitorIndicesParameter.h"
+
+using namespace rfb;
+static LogWriter vlog("MonitorIndicesParameter");
+
+MonitorIndicesParameter::MonitorIndicesParameter(const char* name_, const char* desc_, const char* v)
+: StringParameter(name_, desc_, v) {}
+
+std::set<int> MonitorIndicesParameter::getParam()
+{
+    bool valid = false;
+    std::set<int> indices;
+    std::set<int> config_indices;
+    std::vector<MonitorIndicesParameter::Monitor> monitors = this->monitors();
+
+    if (monitors.size() <= 0) {
+        vlog.error(_("Failed to get monitors."));
+        return indices;
+    }
+
+    valid = parse_indices(value, config_indices);
+    if (!valid) {
+        return indices;
+    }
+
+    if (config_indices.size() <= 0) {
+        return indices;
+    }
+
+    // Go through the monitors and see what indices are present in the config.
+    for (int i = 0; i < ((int) monitors.size()); i++) {
+        if (std::find(config_indices.begin(), config_indices.end(), i) != config_indices.end())
+            indices.insert(monitors[i].fltk_index);
+    }
+
+    return indices;
+}
+
+bool MonitorIndicesParameter::setParam(const char* value)
+{
+    std::set<int> indices;
+
+    if (strlen(value) <= 0)
+        return false;
+
+    if (!parse_indices(value, indices)) {
+        vlog.error(_("Parsing failed for FullScreenSelectedMonitors."));
```

Comment:
Let's assume it's not the code that's buggy and instead state that it's the value that is invalid. :)

Also, it would be helpful to specify _why_ it is invalid.

Reply:
I tried adding some more details about why the parsing went wrong.